alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-input-v2.component.ts

```diff
@@ -0,0 +1,483 @@
+    const uuid = this.route.snapshot.paramMap.get('uuid');
+    const index = this.route.snapshot.paramMap.get('index');
+    this.index = index ? parseInt(index) : 1;
+
+    if (uuid) {
+      this.uuid = uuid;
+    }
+
+    if (fileNumber) {
+      this.fileNumber = fileNumber;
+    }
+  }
+
+  private setYear() {
+    const year = moment('1974');
+    const currentYear = moment().year();
+    while (year.year() <= currentYear) {
+      this.resolutionYears.push(year.year());
+      year.add(1, 'year');
+    }
+    this.resolutionYears.reverse();
+  }
+
+  ngOnDestroy(): void {
+    this.decisionService.clearDecision();
+    this.decisionService.clearDecisions();
+    this.$destroy.next();
+    this.$destroy.complete();
+  }
+
+  async loadData() {
+    if (this.uuid) {
+      await this.decisionService.loadDecision(this.uuid);
+    }
+
+    await this.decisionService.loadDecisions(this.fileNumber);
+
+    this.codes = await this.decisionService.fetchCodes();
+    this.outcomes = this.codes.outcomes;
+
+    await this.prepareDataForEdit();
+  }
+
+  private async prepareDataForEdit() {
+    this.decisionService.$decision
+      .pipe(takeUntil(this.$destroy))
+      .pipe(combineLatestWith(this.modificationService.$modifications, this.decisionService.$decisions))
+      .subscribe(([decision, modifications, decisions]) => {
+        if (decision) {
+          this.existingDecision = decision;
+          this.uuid = decision.uuid;
+        }
+
+        this.mapPostDecisionsToControls(modifications, this.existingDecision);
+
+        if (this.existingDecision) {
+          this.patchFormWithExistingData(this.existingDecision);
+
+          if (decisions.length > 0) {
+            let minDate = null;
+            this.isFirstDecision = true;
+
+            for (const decision of decisions) {
+              if (!minDate && decision.date) {
+                minDate = decision.date;
+              }
+
+              if (minDate && decision.date && minDate > decision.date) {
+                minDate = decision.date;
+              }
+
+              if (
+                this.existingDecision.createdAt > decision.createdAt &&
+                this.existingDecision.uuid !== decision.uuid
+              ) {
+                this.isFirstDecision = false;
+              }
+            }
+
+            if (minDate) {
+              this.minDate = new Date(minDate);
+            }
+
+            if (!this.isFirstDecision) {
+              this.form.controls.postDecision.addValidators([Validators.required]);
+              this.form.controls.decisionMaker.disable();
+            }
+          } else {
+            this.isFirstDecision = true;
+          }
+        } else {
+          this.resolutionYearControl.enable();
+        }
+      });
+  }
+
+  private mapPostDecisionsToControls(
+    modifications: NoticeOfIntentModificationDto[],
+    existingDecision?: NoticeOfIntentDecisionDto
+  ) {
+    //TOOD: Clean this up
```

Comment:
nitpick: TODO may not show up due to typo

Reply:
Cleaned up and removed TODO